### PR TITLE
drivers: serial: added minimal uart driver for esp32c3

### PR DESCRIPTION
**Pull request by uLipe**
Based on uart rom functions, also enable console driver for esp32c3 based board.

* Logging subsystem is now available for esp32c3;
* Core dump is now working when fatal handler is invoked;
* Driver can be used as general purpose uart polled driver.

Original PR: https://github.com/zephyrproject-rtos/zephyr/pull/37316

### CODE REVIEW

**Review by glaubermaroto — APPROVED** [review]

**Review by sylvioalves — APPROVED** [review]

**Review by dcpleung — APPROVED** [review]

File: drivers/serial/uart_rom_esp32c3.c [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+		    &uart_rom_esp32c3_init,	\
+		    NULL,	\
+		    NULL,	\
+		    NULL,	\
+		    PRE_KERNEL_1,	\
+		    CONFIG_KERNEL_INIT_PRIORITY_DEVICE,	\
+		    &uart_rom_esp32c3_api);	\
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(uart0), okay)
+ESP32C3_ROM_UART_INIT(0);
```
Comment (MaureenHelm) [inline drivers/serial/uart_rom_esp32c3.c:60]: Why aren't you using `DT_INST_FOREACH_STATUS_OKAY`?

File: boards/riscv/esp32c3_devkitm/esp32c3_devkitm.dts [hunk 2]
```diff
@@ -12,7 +12,18 @@
 	model = "esp32c3_devkitm";
 	compatible = "espressif,esp32c3";
 
+	aliases {
+		uart-0 = &uart0;
```
Comment (MaureenHelm) [inline boards/riscv/esp32c3_devkitm/esp32c3_devkitm.dts:16]: If you use `DT_INST_FOREACH_STATUS_OKAY` in the driver, then you can remove this alias.

File: dts/riscv/espressif/esp32c3.dtsi [hunk 3]
```diff
@@ -21,12 +21,18 @@
 		#address-cells = <1>;
 		#size-cells = <1>;
 		compatible = "simple-bus";
-		ranges;
```
Comment (MaureenHelm) [inline dts/riscv/espressif/esp32c3.dtsi:24]: Is this change related to uart?

Comment (uLipe) [reply dts/riscv/espressif/esp32c3.dtsi:24]: Oh my bad, it was imported as garbage from other workspace.

File: drivers/serial/uart_rom_esp32c3.c (hunk 1 above)
Comment (uLipe) [reply drivers/serial/uart_rom_esp32c3.c]: Updated, thank you!

File: boards/riscv/esp32c3_devkitm/esp32c3_devkitm.dts (hunk 2 above)
Comment (uLipe) [reply boards/riscv/esp32c3_devkitm/esp32c3_devkitm.dts]: updated, thank you!

**Review by MaureenHelm — APPROVED** [review]
